### ISSUE TITLE
Beckhoff: Implement AUTO_ONOFF

### ISSUE THREAD

**Issue by DominicOram** [ISISComputingGroup/IBEX#6839]
As a user of a Beckhoff controller I would like the option to set my axes to automatically energise/de-energise between moves or  to set them to always be on. This functionality is already there in the GALIL as AUTO_ONOFF. The Beckhoff variable for setting whether the motor is on or not is BENABLE.

### Acceptance Criteria
- [ ] The user can set OFF_CMD to turn the Beckhoff axis off
- [ ] The user can set ON_CMD to turn the Beckhoff axis on
- [ ] The user can set AUTOONOFF_CMD to 1 and the Beckhoff axis will turn on before a move and off again after
- [ ] The user can set AUTOONOFF_CMD to 0 and the on/off status of the Beckhoff axis will be unchanged by a move
- [ ]  AUTOONOFF_STATUS and ON_STATUS also exist to show current device status
- [ ] These commands, and readbacks are available on the OPI, like with the galil

**Comment by FreddieAkeroyd** [comment]
Just to clarify, `BENABLE` is the on/off variable to set, but the "auto" feature needs to be done in our driver? Also is `BENABLE` read/write i.e. if the motor is turned off via anther route will it update to reflect the current status?

**Comment by DominicOram** [comment]
Yes, currently we need to do the auto bit ourselves, I'll talk to Simon about whether it makes sense to put it in the PLC. Yes, it's read/write.

**Comment by rerpha** [comment]
So far what i've got is: 

https://github.com/ISISComputingGroup/EPICS-tcIoc/pull/10
https://github.com/ISISComputingGroup/EPICS-ioc/pull/651

**Comment by DominicOram** [comment]
This looks like it would work if it wasn't for https://github.com/ISISComputingGroup/IBEX/issues/6874

**Comment by rerpha** [comment]
#6874 has solved DMOV and MOVN i believe so this may work, i'll test and put in review if it does

**Comment by rerpha** [comment]
ok, still needs some work, im going to shelve for now in favour of other work

**Comment by rerpha** [comment]
this is going to be done on the beckhoff itself now. 

https://github.com/ISISComputingGroup/EPICS-motorExtensions/pull/86
https://github.com/ISISComputingGroup/EPICS-ioc/pull/969
https://github.com/ISISComputingGroup/IBEX/pull/8966

**Comment by Tom-Willemsen** [comment]
I did not functionally test this locally, but am aware that @rerpha tested it against a real beckhoff on NDXMOTION.
